File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        bool isNeg = false;
        long long y = x;
        
        if(x < 0){
            isNeg = true;
            y = y * -1;
        }
        vector<string> digs;
        
        while(y > 0){
            digs.push_back(to_string(y%10));
            y = y / 10;
        }
        
        long long t = pow(10, digs.size());
        long long res = 0;
        int n = digs.size() - 1;
        
        for(int i = 0; i < digs.size(); i++){
            // cout<<stoi(digs[i])*pow(10, n)<<" ";
            res += stoi(digs[i])*pow(10, n);
            n--;
        }
        
        if(isNeg){
            res = -res;
        }
        if(res > 2147483648 - 1 || res < -2147483648){
                return 0;
        }
        return res;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        isNeg = False

        if x < 0:
            isNeg = True
            x = -x

        while x:
            res *= 10
            res += x%10
            x = x//10
        
        if res > 2**31 - 1:
            return 0
        return -res if isNeg else res